cdap-docs/admin-manual/source/upgrading/packages.rst

```diff
@@ -0,0 +1,182 @@
+.. meta::
+    :author: Cask Data, Inc.
+    :copyright: Copyright © 2016 Cask Data, Inc.
+
+.. _admin-upgrading-packages:
+
+=============================
+Upgrading CDAP using Packages
+=============================
+
+.. _admin-upgrading-packages-cdap:
+
+Upgrading CDAP
+==============
+When upgrading an existing CDAP installation from a previous version, you will need
+to make sure the CDAP table definitions in HBase are up-to-date.
+
+These steps will stop CDAP, update the installation, run an upgrade tool for the table definitions,
+and then restart CDAP.
+
+**These steps will upgrade from CDAP 3.2.x to 3.3.x.** If you are on an earlier version of CDAP,
+please follow the upgrade instructions for the earlier versions and upgrade first to 3.2.x before proceeding.
+
+.. highlight:: console
+
+1. Stop all flows, services, and other programs in all your applications.
+
+#. Stop all CDAP processes::
+
+     $ for i in `ls /etc/init.d/ | grep cdap` ; do sudo service $i stop ; done
+
+#. Update the CDAP file definition lists by running either of these methods:
+ 
+   - On RPM using Yum:
+
+     .. include:: ../_includes/installation/installation.txt 
+        :start-after: Download the Cask Yum repo definition file:
+        :end-before:  .. end_install-rpm-using-yum
+
+   - On Debian using APT:
+
+     .. include:: ../_includes/installation/installation.txt 
+        :start-after: Download the Cask APT repo definition file:
+        :end-before:  .. end_install-debian-using-apt
+
+#. Update the CDAP packages by running either of these methods:
+
+   - On RPM using Yum (on one line)::
+
+       $ sudo yum install cdap cdap-gateway \
+             cdap-hbase-compat-0.96 cdap-hbase-compat-0.98 cdap-hbase-compat-1.0 \
+             cdap-hbase-compat-1.0-cdh cdap-hbase-compat-1.1 \
+             cdap-kafka cdap-master cdap-security cdap-ui
+
+   - On Debian using APT (on one line)::
+
+       $ sudo apt-get install cdap cdap-gateway \
+             cdap-hbase-compat-0.96 cdap-hbase-compat-0.98 cdap-hbase-compat-1.0 \
+             cdap-hbase-compat-1.0-cdh cdap-hbase-compat-1.1 \
+             cdap-kafka cdap-master cdap-security cdap-ui
+
+#. If you are upgrading a secure Hadoop cluster, you should authenticate with ``kinit``
+   as the user that runs CDAP Master (the CDAP user)
+   before the next step (the running of the upgrade tool)::
+
+     $ kinit -kt <keytab> <principal>
+
+#. Run the upgrade tool, as the user that runs CDAP Master (the CDAP user)::
+
+     $ /opt/cdap/master/bin/svc-master run co.cask.cdap.data.tools.UpgradeTool upgrade
+     
+   Note that once you have upgraded an instance of CDAP, you cannot reverse the process; down-grades
+   to a previous version are not possible.
+   
+   The Upgrade Tool will produce output similar to the following, prompting you to continue with the upgrade:
+   
+    .. container:: highlight
+
+      .. parsed-literal::    
+    
+        UpgradeTool - version |short-version|-xxxxx.
```

Comment:
The actual output appears as
```
UpgradeTool - version 3.1.1-1439543165346.
```
so I shortened it and indicated that it had been changed.